### PR TITLE
feat(): previews template customization

### PR DESCRIPTION
**Pull request by locinus**
PR related to git issue: https://github.com/Ks89/angular-modal-gallery/issues/243
Implementing the solution proposed in: https://github.com/Ks89/angular-modal-gallery/issues/243#issuecomment-1024634334

### CODE REVIEW

File: projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts [hunk 1]
```diff
@@ -194,6 +203,9 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
    * @param Action action that triggered the source event or `Action.NORMAL` if not specified
    */
   onImageEvent(preview: InternalLibImage, event: KeyboardEvent | MouseEvent, action: Action = Action.NORMAL): void {
+    // It's suggested to stop propagation of the event, so the
```
Comment (Ks89) [inline projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts:206]: Why is this necessary? I implemented all other features without this.
Did you find a specific reference about this somewhere?

File: src/app/modal-gallery/modal-gallery.component.ts [hunk 2]
```diff
@@ -512,7 +520,8 @@ export class ModalGalleryExampleComponent implements OnDestroy {
       id,
       images: imagesArrayToUse,
       currentImage: imageToShow,
-      libConfig
+      libConfig,
+      previewsTemplate: this.previewsTemplate,
```
Comment (Ks89) [inline src/app/modal-gallery/modal-gallery.component.ts:524]: previewsTemplate should be applied only to a specific example, otherwise all examples in this demo app will use that template

Comment (Ks89) [reply src/app/modal-gallery/modal-gallery.component.ts:524]: I suggest, for instance to add a new exaperimental example at the end on the page with a dedicare "openModalWithPreviewsTemplate" or something similar

**Review by Ks89 — CHANGES_REQUESTED** [review]
I added some comments.

I executed this PR on my pc and there is a weird flickering when clicking on previews.
If you open the gallery clicking on the first image, then on the second preview, there is a strange behaviour. This doesn't happen in the stable version.

**Comment by locinus** [comment]
I will check for any flickering, and see what's in cause. I will work in that in a couple of days.

File: projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts (hunk 1 above)
Comment (locinus) [reply projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts:206]: I believe it's necessary because the CDK overlay below is awaiting for any click to close the whole gallery, and receives it when the user clicks on the description text for instance (at least on Windows Chrome/Firefox). I applied it on the whole template-surrounding div here, because most likely a click on that zone is expected to lead to the connected preview and not to close the gallery unintentionally.

File: src/app/modal-gallery/modal-gallery.component.ts (hunk 2 above)
Comment (locinus) [reply src/app/modal-gallery/modal-gallery.component.ts:524]: Sure, I set it quickly here for the sake of the discussion. It'll have its own example when I finalize the PR.

**Comment by locinus** [comment]
Just a heads up: very busy these days; I'll look at it by next week. Sorry for the delay !

**Comment by locinus** [comment]
Hey there. So I couldn't reproduce the flickering effect. Does it happen on the preview image and description? or on any other element on the page?
I'll try to get as much as I can to your environment (mine's Windows + Chrome/FF). Yours' MacOS, is it? With Safari or some other browsers?

**Comment by Ks89** [comment]
I'm using macOS with Chrome. Both at the latest versions.
Check the video attached to this post

https://user-images.githubusercontent.com/6057207/158876430-a11d430f-ed99-487c-9613-66c27e98b8d9.mov

**Comment by locinus** [comment]
Oh nice! Thanks for the video.
Okay, so the flickering is within the previews, which should stay still, but "flicker" when you click on the second thumbnail, is that it?

Now - twist! - if this is the flickering you're talking about, I think I can reproduce it, on my PR but also on your demo site, especially when loading the site fresh again with cache disabled and all.
Check this GIF (Windows/Chrome):
ks89-flickering
I can also observe a kind of "flickering rearrangement" when I come back to the picture before the last after reaching the end.

Or is it something else you were adressing?
In all cases, I'd be up to try to arrange this, but I'd need to know if it's related to the PR or not.
All the best!

**Comment by Ks89** [comment]
Oh damn I forgot to answer. I'll check this evening.
However the doc website is not a good benchmark tool because github pages is super slow.
A good comparison is testing master and you pr locally with the demo application included inside the repo

**Comment by locinus** [comment]
Hello!
So I tested again with local demo (yarn start), with the 900 repo I forked and with my PR.
From my side, there's no noticeable differences: if no cache is used, the flickering does happen in both version; if cache is used, it's alright (yet flickering might occur from time to time), also in both versions.

900 forked, cache
900_local_cache

900 forked, no cache
900_local_NO_cache

PR, cache
PR_local_cache

PR, no cache
PR_local_NO_cache

**Comment by Ks89** [comment]
Yes, you are right. There is a small delay with no cache, because it needs to download the file.
Also on master branch, it happens the same with no cache.

Since now v.9.0.0 is out, I suggest to refer to develop branch.
Branch 900 should be removed, because obsolete. I used that branch only to develop 900 before the official release.
If everything about this PR is backward compatible, it can be merged into develop to prepare v9.1.0.

**Comment by locinus** [comment]
Yes, ok! I'll polish the PR with the examples tidied up and all.

As for the clipping, I think I'll have a look to it if I have time. I open another thread on the topic !
